The following modules were not unloaded:
  (Use "module --force purge" to unload all):

  1) Stages/2025

The following have been reloaded with a version change:
  1) sympy/1.13.3 => sympy/1.13.1

WARNING:__main__:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
[W731 09:10:04.867587267 socket.cpp:752] [c10d] The client socket cannot be initialized to connect to [jrc0258i.jureca]:7010 (errno: 97 - Address family not supported by protocol).
[W731 09:10:13.318115866 socket.cpp:752] [c10d] The client socket cannot be initialized to connect to [jrc0258i.jureca]:7010 (errno: 97 - Address family not supported by protocol).
[W731 09:10:13.318131386 socket.cpp:752] [c10d] The client socket cannot be initialized to connect to [jrc0258i.jureca]:7010 (errno: 97 - Address family not supported by protocol).
[W731 09:10:13.318157446 socket.cpp:752] [c10d] The client socket cannot be initialized to connect to [jrc0258i.jureca]:7010 (errno: 97 - Address family not supported by protocol).
[W731 09:10:13.318158536 socket.cpp:752] [c10d] The client socket cannot be initialized to connect to [jrc0258i.jureca]:7010 (errno: 97 - Address family not supported by protocol).
/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py:122: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  model.load_state_dict(torch.load(model_path, map_location=device))
/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py:122: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  model.load_state_dict(torch.load(model_path, map_location=device))
/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py:122: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  model.load_state_dict(torch.load(model_path, map_location=device))
/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py:122: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  model.load_state_dict(torch.load(model_path, map_location=device))
[rank3]: Traceback (most recent call last):
[rank3]:   File "/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py", line 166, in <module>
[rank3]:     main()
[rank3]:   File "/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py", line 122, in main
[rank3]:     model.load_state_dict(torch.load(model_path, map_location=device))
[rank3]:   File "/p/software/default/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/nn/modules/module.py", line 2516, in load_state_dict
[rank3]:     raise TypeError(
[rank3]: TypeError: Expected state_dict to be dict-like, got <class 'torch.nn.parallel.distributed.DistributedDataParallel'>.
[rank1]: Traceback (most recent call last):
[rank1]:   File "/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py", line 166, in <module>
[rank1]:     main()
[rank1]:   File "/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py", line 122, in main
[rank1]:     model.load_state_dict(torch.load(model_path, map_location=device))
[rank1]:   File "/p/software/default/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/nn/modules/module.py", line 2516, in load_state_dict
[rank1]:     raise TypeError(
[rank1]: TypeError: Expected state_dict to be dict-like, got <class 'torch.nn.parallel.distributed.DistributedDataParallel'>.
[rank2]: Traceback (most recent call last):
[rank2]:   File "/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py", line 166, in <module>
[rank2]:     main()
[rank2]:   File "/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py", line 122, in main
[rank2]:     model.load_state_dict(torch.load(model_path, map_location=device))
[rank2]:   File "/p/software/default/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/nn/modules/module.py", line 2516, in load_state_dict
[rank2]:     raise TypeError(
[rank2]: TypeError: Expected state_dict to be dict-like, got <class 'torch.nn.parallel.distributed.DistributedDataParallel'>.
[rank0]: Traceback (most recent call last):
[rank0]:   File "/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py", line 166, in <module>
[rank0]:     main()
[rank0]:   File "/p/project1/training2529/marxen1/amm-risk-lab/python/ml/PLV/scripts/run_ddp_finetuning.py", line 122, in main
[rank0]:     model.load_state_dict(torch.load(model_path, map_location=device))
[rank0]:   File "/p/software/default/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/nn/modules/module.py", line 2516, in load_state_dict
[rank0]:     raise TypeError(
[rank0]: TypeError: Expected state_dict to be dict-like, got <class 'torch.nn.parallel.distributed.DistributedDataParallel'>.
[rank0]:[W731 09:10:16.046091759 ProcessGroupNCCL.cpp:1250] Warning: WARNING: process group has NOT been destroyed before we destruct ProcessGroupNCCL. On normal program exit, the application should call destroy_process_group to ensure that any pending NCCL operations have finished in this process. In rare cases this process can exit before this point and block the progress of another member of the process group. This constraint has always been present,  but this warning has only been added since PyTorch 2.4 (function operator())
W0731 09:10:17.833000 940317 /p/software/fs/jurecadc/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/distributed/elastic/multiprocessing/api.py:897] Sending process 940375 closing signal SIGTERM
W0731 09:10:17.841000 940317 /p/software/fs/jurecadc/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/distributed/elastic/multiprocessing/api.py:897] Sending process 940376 closing signal SIGTERM
W0731 09:10:17.848000 940317 /p/software/fs/jurecadc/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/distributed/elastic/multiprocessing/api.py:897] Sending process 940377 closing signal SIGTERM
E0731 09:10:18.068000 940317 /p/software/fs/jurecadc/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/distributed/elastic/multiprocessing/api.py:869] failed (exitcode: 1) local_rank: 0 (pid: 940374) of binary: /p/home/jusers/marxen1/jureca/course/sc_venv_template/venv/bin/python3
Traceback (most recent call last):
  File "/p/home/jusers/marxen1/jureca/course/sc_venv_template/venv/bin/torchrun_jsc", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/p/project1/training2529/marxen1/amm-risk-lab/envs/jureca0/venv/lib/python3.12/site-packages/torchrun_jsc/__init__.py", line 44, in main
    run_main()
  File "/p/project1/training2529/marxen1/amm-risk-lab/envs/jureca0/venv/lib/python3.12/site-packages/torchrun_jsc/run.py", line 91, in main
    runpy.run_module('torch.distributed.run', run_name='__main__')
  File "<frozen runpy>", line 229, in run_module
  File "<frozen runpy>", line 88, in _run_code
  File "/p/software/default/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/distributed/run.py", line 923, in <module>
    main()
  File "/p/software/default/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 355, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/p/software/default/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/distributed/run.py", line 919, in main
    run(args)
  File "/p/software/default/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/distributed/run.py", line 910, in run
    elastic_launch(
  File "/p/software/default/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/distributed/launcher/api.py", line 138, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/p/software/default/stages/2025/software/PyTorch/2.5.1-gcccoreflexiblas-13.3.0-3.4.4/lib/python3.12/site-packages/torch/distributed/launcher/api.py", line 269, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
python/ml/PLV/scripts/run_ddp_finetuning.py FAILED
------------------------------------------------------------
Failures:
  <NO_OTHER_FAILURES>
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2025-07-31_09:10:17
  host      : jrc0258.jureca
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 940374)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
srun: error: jrc0258: task 0: Exited with exit code 1
